tests/api/test_tasks.py

```diff
@@ -0,0 +1,108 @@
+#
+# This file is part of RERO ILS.
+# Copyright (C) 2017 RERO.
+#
+# RERO ILS is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# RERO ILS is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with RERO ILS; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, RERO does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Tasks tests."""
+
+import json
+from datetime import datetime, timedelta, timezone
+
+from flask import url_for
+from invenio_accounts.testutils import login_user_via_session
+from invenio_circulation.search.api import LoansSearch
+from utils import flush_index, get_json
+
+from rero_ils.modules.loans.api import Loan, LoanAction, get_due_soon_loans, \
+    get_overdue_loans
+from rero_ils.modules.notifications.api import NotificationsSearch, \
+    number_of_reminders_sent
+from rero_ils.modules.notifications.tasks import \
+    create_over_and_due_soon_notifications
+
+
+def test_create_over_and_due_soon_notifications_task(
+        client, librarian_martigny_no_email, patron_martigny_no_email,
+        item_lib_martigny, circ_policy_short_martigny):
+    """Test overdue and due_soon loans."""
+    login_user_via_session(client, librarian_martigny_no_email.user)
+    item = item_lib_martigny
+    item_pid = item.pid
+    patron_pid = patron_martigny_no_email.pid
+    # checkout
+    res = client.post(
+        url_for('api_item.checkout'),
+        data=json.dumps(
+            dict(
+                item_pid=item_pid,
+                patron_pid=patron_pid
+            )
+        ),
+        content_type='application/json',
+    )
+    assert res.status_code == 200
+
+    data = get_json(res)
+    loan_pid = data.get('action_applied')[LoanAction.CHECKOUT].get('loan_pid')
+    loan = Loan.get_record_by_pid(loan_pid)
+
+    # test due_soon notification
+    end_date = datetime.now(timezone.utc) + timedelta(days=3)
+    loan['end_date'] = end_date.isoformat()
+    loan.update(loan, dbcommit=True, reindex=True)
+
+    due_soon_loans = get_due_soon_loans()
+
+    assert due_soon_loans[0].get('loan_pid') == loan_pid
+
+    create_over_and_due_soon_notifications.delay()
+    flush_index(NotificationsSearch.Meta.index)
```

Comment:
should not be needed. This can be done by `create_over_and_due_soon_notifications`